src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -204,3 +210,18 @@ def check_cli(cmd):
         print('CLI self-test completed: OK')
     else:
         raise CLIError(exceptions)
+
+
+def _remove_adal_token_cache():
+    """Remove ADAL token cache file ~/.azure/accessTokens.json, as it is no longer needed by MSAL-based Azure CLI.
+    """
+    from azure.cli.core._environment import get_config_dir
+    adal_token_cache = os.path.join(get_config_dir(), 'accessTokens.json')
+    try:
+        os.remove(adal_token_cache)
+        logger.warning("ADAL token cache '%s' has been deleted. `az login` no longer generates this file. "
+                       "For more details, see https://docs.microsoft.com/cli/azure/msal-based-azure-cli",
+                       adal_token_cache)
```

Comment:
Wouldn't such a warning too confusing and scary to normal users? I think it would be fine to just remove it silently. But it is up to you. I'm still approving this PR.

Reply:
@chasewilson, thoughts?

Reply:
Hmm, would it be a good idea to use logging.info? This would at least provide the information if someone was debugging or using verbose to get more info and for customers using logging would provide a record and reason for the deletion?

Reply:
> Hmm, would it be a good idea to use logging.info? This would at least provide the information if someone was debugging or using verbose to get more info and for customers with logging enabled would provide a record and reason for the deletion?

Makes sense. Could also consider `logging.debug`, then. I am fine either way.

Reply:
This is a one-time execution. After that, this warning won't be shown anymore, so it will get lost forever! 😆

People usually only use `--verbose` and `--debug` after hitting a failure.

Reply:
> This is a one-time execution. After that, this warning won't be shown anymore, so it will get lost forever! 😆
> 
> People usually only use `--verbose` and `--debug` after hitting a failure.

Personally, if I run a product and it emits an unfamiliar but seemingly scary message (all warnings are scary), I would be alerted and then try to repro and see if I need to report back. But then, if my repro attempt can not reproduce that warning, I would be very confuse. This feeling of unpredictability would decrease my trust to the product. (For what it's worth, this was the same thought process when I was testing a new security camera that I bought. Now I'm considering to return it. :-) )

In short, let's at least make it an INFO or DEBUG, not a WARNING. After all, it is an almost expected behavior during `az logout` or `az account clean`, so, there is nothing serious need to be warned.

Reply:
We're adding a new concept here (ADAL) that most customers won't know about.  Maybe saying "Legacy ADAL Token Cache has been cleaned up" to show it's normal and something they probably won't need to act on. If they're actually using it directly, they'll know what we're talking about ?

Reply:
> We're adding a new concept here (ADAL) that most customers won't know about.

This makes sense. Maybe let's just remove the log and save us and users from the fuss.